ibmdbpy/learn/association_rules.py

```diff
@@ -541,23 +541,19 @@ def _retrieve_AssociationRules_Model(self, modelname, verbose=False):
         # In case the implementation of the IDA method changes, this may break
         # But still would not be difficult to fix 
 
-        assocpatterns = self._idadb.ida_query('SELECT * FROM "' +
-        self._idadb.current_schema + '"."' + modelname + '_ASSOCPATTERNS"')
+        assocpatterns = self._idadb.ida_query('SELECT * FROM ' + modelname + '_ASSOCPATTERNS')
         assocpatterns.columns = ["ITEMSETID","ITEMID"]
         assocpatterns.columns = [x.upper() for x in assocpatterns.columns]
 
-        assocpatterns_stats = self._idadb.ida_query('SELECT * FROM "' +
-        self._idadb.current_schema + '"."' + modelname + '_ASSOCPATTERNS_STATISTICS"')
+        assocpatterns_stats = self._idadb.ida_query('SELECT * FROM ' + modelname + '_ASSOCPATTERNS_STATISTICS')
```

Comment:
Model names can include only alphanumeric characters and underscores because they have to pass the checks of  ibmddbpy.utils.check_tablename which turns lowercase to uppercase characters, 
for instance see
https://github.com/ibmdbanalytics/ibmdbpy/blob/7019ae3829874e800782ea398ec400f4dce3664e/ibmdbpy/learn/association_rules.py#L229-L235
Therefore it is not necessary to include the model schema and model name inside double quotes.

Reply:
Let's not focus on upper and lower case for now, what about model names containing spaces? I suppose that's supported, at least I couldn't find any documentation claiming otherwise (for example https://www.ibm.com/support/knowledgecenter/SS6NHC/com.ibm.swg.im.dashdb.analytics.doc/doc/r_model_names_examples.html doesn't add any restrictions, https://www.ibm.com/support/knowledgecenter/SS6NHC/com.ibm.swg.im.dashdb.analytics.doc/doc/r_model_component_naming_conventions.html even talks about a model called `Sample Model`).

Reply:
idmdbpy is more restrictive concerning model names than the analytic stored procedures. The same is true, by the way, for table names.
It supports only modelnames with alphanumeric characters and underscores. 
This checked is performed by` ibmdbpy.utils.check_tablename(modelname)`, for instance for exists_model in:
https://github.com/ibmdbanalytics/ibmdbpy/blob/7019ae3829874e800782ea398ec400f4dce3664e/ibmdbpy/base.py#L539-L541
...
https://github.com/ibmdbanalytics/ibmdbpy/blob/7019ae3829874e800782ea398ec400f4dce3664e/ibmdbpy/base.py#L564-L566

So to make that restriction clear I would describe it in the ibmdbpy documentation, like it is done in the docstring for IdaDataBase.rename:
https://github.com/ibmdbanalytics/ibmdbpy/blob/7019ae3829874e800782ea398ec400f4dce3664e/ibmdbpy/base.py#L1026-L1038

Reply:
Thanks for digging up that information, looks like the point about whitespace characters is moot.